### PR TITLE
Reorder

### PR DESCRIPTION
**Pull request by XKTZ**
# Pull Request Checklist

## Reference Issue

This is a superset of issue [Top Down](https://github.com/castorini/ura-projects/issues/46). This PR reorganized the various reordering methods including sliding window, top down, as well as ListT5's tournament sort methodology. Now it is allowed to use command like `--reorder_policy="top_down:{\"top_k\": 10, \"pivot\": ${PIVOT}, \"shuffle\": true, \"r\": 1}"` to specify a reorderer.

ref: 

## Checklist Items

Before submitting your pull request, please review these items:

- [*] Have you followed the [contributing guidelines](CONTRIBUTING.md)?
- [*] Have you verified that there are no existing [Pull Requests](https://github.com/castorini/rank_llm/pulls) for the same update/change?
- [ ] Have you updated any relevant documentation or added new tests where needed?

# PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [*] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... 
    - Description:

### CODE REVIEW

File: src/rank_llm/scripts/run_rank_llm.py [hunk 1]
```diff
@@ -161,8 +166,8 @@ def main(args):
     parser.add_argument(
         "--step_size",
         type=int,
-        default=10,
-        help="step size for the sliding window approach",
+        default=20,
+        help="window size for the LLM",
     )
```
Comment (ronakice) [inline src/rank_llm/scripts/run_rank_llm.py:171]: step and window are different?

File: README.md [hunk 2]
```diff
@@ -147,6 +147,18 @@ Omit `--use_logits` if you wish to perform traditional listwise reranking.
...
+
+* Other LLM reranker method, like 
+
+```
+python src/rank_llm/scripts/run_rank_llm.py  --model_path=castorini/rank_zephyr_7b_v1_full --top_k_candidates=100 --dataset=dl20 \
+    --retrieval_method=SPLADE++_EnsembleDistil_ONNX --prompt_mode=rank_GPT  --context_size=4096 --variable_passages \
+    --reorder_policy="tournament_sort:{top_k: 10, r: 1}"
+```
+
+### Run end to end - Rank Zephyr with [Top Down](https://)
```
Comment (ronakice) [inline README.md:160]: @XKTZ what is this?

File: README.md [hunk 3]
```diff
@@ -283,6 +295,34 @@ If you would like to cite the FIRST methodology, please consider citing:
 }
 ```
 
+If you woud like to cite the ListT5's tournament sort methodology, please consider citing
```
Comment (ronakice) [inline README.md:298]: I don't think this is the first instance of tournament sort, something has probably been done before, maybe we drop it for now
